### PR TITLE
Adding support for new ocp_version_ranges

### PR DESCRIPTION
**Pull request by lipoja**
[CLOUDDST-21988]

### CODE REVIEW

File: iib/workers/tasks/build_merge_index_image.py [hunk 1]
```diff
@@ -440,43 +440,73 @@ def is_bundle_version_valid(
     v4.7   | included  | NOT included |   included   |  included   |  included
     -------------------------------------------------------------------------------
     v4.8   | included  | NOT included | NOT included |  included   |  included
+
+
+           |  "=v4.5|=v4.6"  |   "=v4.5|>=v4.7"   |   "<=v4.7"   |  "<=v4.6|>=v4.8"
```
Comment (yashvardhannanavati) [inline iib/workers/tasks/build_merge_index_image.py:445]: I don't think `"<=v4.6|>=v4.8"` is a supported version in Pyxis.

File: iib/workers/tasks/build_merge_index_image.py [hunk 2]
```diff
@@ -440,43 +440,73 @@ def is_bundle_version_valid(
...
+                    return True
+            # MYPY error: Unsupported right operand type for in ("Optional[str]")
+            elif '-' in bundle_version_label:  # type: ignore
+                min_version, max_version = [
+                    Version(version) for version in bundle_version.split('-')
+                ]
+                if min_version <= ocp_version <= max_version:
+                    return True
+            # MYPY error: Unsupported right operand type for in ("Optional[str]")
+            elif bundle_version_label.startswith('>='):  # type: ignore
```
Comment (yashvardhannanavati) [inline iib/workers/tasks/build_merge_index_image.py:490]: this approach would also make only `>=4.15` also valid, no?

**Review by yashvardhannanavati — CHANGES_REQUESTED** [review]

File: iib/workers/tasks/build_merge_index_image.py (hunk 1 above)
Comment (lipoja) [reply iib/workers/tasks/build_merge_index_image.py:445]: should I remove it?

Comment (yashvardhannanavati) [reply iib/workers/tasks/build_merge_index_image.py:445]: Yeah, instead =v4.6|=v4.7|>=v4.8 (when v4.8 is not deprecated) should be there, no?

File: iib/workers/tasks/build_merge_index_image.py (hunk 2 above)
Comment (yashvardhannanavati) [reply iib/workers/tasks/build_merge_index_image.py:490]: I don't think only `>=v4.15` works in Pyxis (https://pyxis.engineering.redhat.com/v1/operators/indices?ocp_versions_range=%3E=v4.11) and neither does `<=v4.x` anymore. I believe the support for the latter ] was dropped some time ago.

File: iib/workers/tasks/build_merge_index_image.py (hunk 1 above)
Comment (lipoja) [reply iib/workers/tasks/build_merge_index_image.py]: @yashvardhannanavati please check

File: iib/workers/tasks/build_merge_index_image.py [hunk 3]
```diff
@@ -440,6 +440,18 @@ def is_bundle_version_valid(
     v4.7   | included  | NOT included |   included   |  included   |  included
     -------------------------------------------------------------------------------
     v4.8   | included  | NOT included | NOT included |  included   |  included
+
+
+           |  "=v4.5|=v4.6"  |   "=v4.5|>=v4.7"   |
```
Comment (chandwanitulsi) [inline iib/workers/tasks/build_merge_index_image.py:445]: This is assuming 4.7 is not EOL right? we should add it somewhere in comment

**Review by chandwanitulsi — APPROVED** [review]
Few comments, other than those LGTM

**Review by yashvardhannanavati — APPROVED** [review]

File: iib/workers/tasks/build_merge_index_image.py [hunk 4]
```diff
@@ -456,8 +468,23 @@ def is_bundle_version_valid(
...
+                    'either "=" or ">="'
+                )
+                return False
+            for version in versions:
+                if version.startswith('>='):  # type: ignore
+                    if ocp_version >= Version(version.strip('>=')):
+                        return True
+                elif version.startswith('='):  # type: ignore
+                    if Version(version.strip('=')) == ocp_version:
+                        return True
```
Comment (xDaile) [inline iib/workers/tasks/build_merge_index_image.py:485]: Is it okay to return True if we did not check all the versions?

File: tests/test_workers/test_tasks/test_build_merge_index_image.py [hunk 5]
```diff
@@ -681,6 +681,17 @@ def test_add_bundles_missing_in_source_none_missing(
...
         ('=v4.5', 'v4.6', False, False),
         ('v4.5-v4.7', 'v4.6', False, True),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.8', False, False),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.9', False, True),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.10', False, True),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.11', False, False),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.12', False, True),
+        ('=v4.9|=v4.10|>=v4.13', 'v4.13', False, True),
+        ('=v4.5|=v4.6', 'v4.6', False, True),
+        ('=v4.5|v4.10', 'v4.12', False, False),
```
Comment (xDaile) [inline tests/test_workers/test_tasks/test_build_merge_index_image.py:691]: Is there missing "=" before "v4.10"?

**Review by xDaile — APPROVED** [review]

File: tests/test_workers/test_tasks/test_build_merge_index_image.py [hunk 6]
```diff
@@ -681,6 +681,17 @@ def test_add_bundles_missing_in_source_none_missing(
     (
         ('=v4.5', 'v4.6', False, False),
         ('v4.5-v4.7', 'v4.6', False, True),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.8', False, False),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.9', False, True),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.10', False, True),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.11', False, False),
+        ('=v4.9|=v4.10|>=v4.12', 'v4.12', False, True),
+        ('=v4.9|=v4.10|>=v4.13', 'v4.13', False, True),
```
[the remark this answers is not in the document]
Comment (lipoja) [reply tests/test_workers/test_tasks/test_build_merge_index_image.py:689]: This will test bot "separately" if it takes ">" and "=" versions - just testing the corner case.
